base/shell/explorer/traydeskbtn.cpp

```diff
@@ -0,0 +1,443 @@
+/*
+ * PROJECT:     ReactOS Explorer
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Show Desktop tray button implementation
+ * COPYRIGHT:   Copyright 2006-2007 Thomas Weidenmueller <w3seek@reactos.org>
+ *              Copyright 2018-2022 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ *              Copyright 2023 Ethan Rodensky <splitwirez@gmail.com>
+ */
+
+#include "traydeskbtn.h"
+#include <limits.h>
+
+#define IDI_SHELL32_DESKTOP 35
+#define IDI_IMAGERES_DESKTOP 110
+
+#define SHOW_DESKTOP_TIMER_ID 999
+#define SHOW_DESKTOP_TIMER_INTERVAL 200
+
+CTrayShowDesktopButton::CTrayShowDesktopButton() :
+    m_nClickedTime(0),
+    m_inset({2, 2}),
+    m_icon(NULL),
+    m_highContrastMode(FALSE),
+    m_drawWithDedicatedBackground(FALSE),
+    m_bHovering(FALSE),
+    m_bPressed(FALSE),
+    IsHorizontal(FALSE)
+{
+}
+
+BOOL
+CTrayShowDesktopButton::GetTaskbar(OUT HWND* taskbarWnd)
+{
+    *taskbarWnd = NULL;
+    auto parent = GetParent();
```

Comment:
"auto" ? You don't know the return type? :rofl:

Reply:
![image](https://github.com/reactos/reactos/assets/34009058/3e3fdf56-e1d3-4b65-bce0-22951f7d8eb7)

Reply:
Then, s/auto/HWND/ :man_shrugging:

Reply:
My bad, it's actually a `CWindow`, not an `HWND`.

I presume you still want me to change `auto` to `CWindow`? That much I've discerned from context...but for future reference, what exactly does that "s/auto/HWND/" thing mean? Is that some kind of notation for...something?

Reply:
Before changing the auto thing, I'm going to have another question for this piece of code.

Regarding the notation `s/old/new/` , this is the famous Vi(m) way of replacing instances of `old` by `new` in text, and since then became an internet "meme"/way to say that intent.

Reply:
Alright, so...what's the question?

Huh...never used Vi(m), so...good to know, I guess?

Reply:
I always assumed this came from `sed`, not `vi`

Reply:
I used a Digital Equipment Corporation (DEC) PDP-9 computer back in 1970 at my University that had a Teletype Model 33 ASR terminal (https://en.wikipedia.org/wiki/Teletype_Model_33). This had a "Line Editor" that was built in. It only allowed working on one line at a time. It used the syntax as shown above. The first 's' was for the search function and this was followed by the '/' to separate 'what to search for' followed by the '/' again to separate 'what to replace it with' and then the closing '/' to end the command. Since you only edited a line at a time, you did not have to worry about this affecting anything but the single line you were working on. So this syntax is quite old based on my experience and precedes any of the later editors that I see mentioned above. I suspect that they duplicated the syntax, because many programmers were familiar with it from earlier times.